packages/router/rollup.config.js

```diff
@@ -13,185 +11,56 @@ const {
 } = require("../../rollup.utils");
 const { name, version } = require("./package.json");
 
-module.exports = function rollup() {
+function getRollupConfig(format, filename, includeTypesAndCopy = false) {
   const { ROOT_DIR, SOURCE_DIR, OUTPUT_DIR } = getBuildDirectories(
     name,
     // We don't live in a folder matching our package name
     "router"
   );
 
-  // JS modules for bundlers
-  const modules = [
-    {
-      input: `${SOURCE_DIR}/index.ts`,
-      output: {
-        file: `${OUTPUT_DIR}/index.js`,
-        format: "esm",
-        sourcemap: !PRETTY,
-        banner: createBanner("@remix-run/router", version),
-      },
-      plugins: [
-        extensions({ extensions: [".ts"] }),
-        babel({
-          babelHelpers: "bundled",
-          exclude: /node_modules/,
-          presets: [
-            ["@babel/preset-env", { loose: true }],
-            "@babel/preset-typescript",
-          ],
-          plugins: ["babel-plugin-dev-expression"],
-          extensions: [".ts"],
-        }),
-        typescript({
-          tsconfig: path.join(__dirname, "tsconfig.json"),
-          exclude: ["__tests__"],
-          noEmitOnError: true,
-        }),
-        copy({
-          targets: [
-            { src: path.join(ROOT_DIR, "LICENSE.md"), dest: SOURCE_DIR },
-          ],
-          verbose: true,
-        }),
-      ].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+  return {
```

Comment:
This new "shared" config is basically just the former ESM config and we pass either `format:'esm'` or `format: 'cjs'`